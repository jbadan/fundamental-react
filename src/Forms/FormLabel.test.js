import FormLabel from './FormLabel';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<FormLabel />', () => {
    const formLabel = (
        <FormLabel forAttr='input-1' required>
            Default Input
        </FormLabel>
    );

    test('create form label', () => {
        let component = renderer.create(formLabel);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the FormLabel component', () => {
            const element = mount(<FormLabel data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
