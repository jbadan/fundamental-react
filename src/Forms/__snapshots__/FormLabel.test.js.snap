// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormLabel /> create form label 1`] = `
<label
  aria-required={true}
  className="fd-form__label"
  forAttr="input-1"
>
  Default Input
  *
</label>
`;
